### PR TITLE
Aptos wallet adapter react

### PR DESCRIPTION
**Pull request by aaronTufts**
A simple example of an Aptos wallet adapter FE that gets user input for a Move call tx and submits that to a BE that builds the tx and sponsors it with Shinami's Gas Station. Currently has two modes from there:

- If a dapp-kit connected wallet, use that. There are four methods, and the dev has to use comments to choose which to run: 1) FE build tx and sign, BE sponsor submit 2) FE build and sign, BE sponsor, FE submit 3) BE build and sponsor, FE sign and submit, 4) BE build and sponsor, FE sign, BE submit.  
- If no connected wallet, uses the Shinami Invisible Wallet API to sponsor, sign, and execute the transaction.

There is a README to get you up and running.

My goal was to show customers a simple but useful example of how to integrate a FE that uses `@aptos-labs/wallet-adapter-react` with a Shinami BE. It wasn't meant to be a starter template for a production app, but to answer basic questions some customers have when they integrate to get them up and going faster. That said, you both obviously have more experience and knowledge about best practices on the FE and BE, so let me know if there's some refactoring or other change you think would make for a better example.

### CODE REVIEW

File: aptos/typescript/wallet_adapter_react/src/client/App.tsx [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+      else {
+        pendingTxResponse = await invisibleWalletTx(message);
+      }
+
+      if(pendingTxResponse?.hash) {
+          waitForTxAndUpdateResult(pendingTxResponse.hash);
+      } else {
+          console.log("Unable to find a digest returned from the backend.");
+      }  
+  } catch (e) {
```
Comment (jasonxh) [inline aptos/typescript/wallet_adapter_react/src/client/App.tsx]: Indentation seems off.

File: aptos/typescript/wallet_adapter_react/src/client/App.tsx [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    const resp = await axios.post('/invisibleWalletTx', {
+      message
+    });
+    return resp.data.pendingTx;
+}
+
+
+  return (
+    <>
+          <h1>Shinami Sponsored Transactions with @aptos-labs/wallet-adapter-react</h1>
```
Comment (jasonxh) [inline aptos/typescript/wallet_adapter_react/src/client/App.tsx]: Same. Please reformat the entire file.

File: aptos/typescript/wallet_adapter_react/src/server/main.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  console.log("Server is listening on port 3000..."),
+);
+
+
+
+// Endpoint to:
+//  1. Build and a feePayer SimpleTransaction with the given user intput.
+//  2. Sign, sponsor, and submit it for an Invisible Wallet sender.
+//  3. Return the PendingTransactionResponse to the FE
+app.post('/invisibleWalletTx', async (req, res, next) => {
```
Comment (jasonxh) [inline aptos/typescript/wallet_adapter_react/src/server/main.ts:75]: Please add a callout that all of these APIs are UNAUTHENTICATED, and they must add their own user auth / session management to secure them.

Comment (aaronTufts) [reply aptos/typescript/wallet_adapter_react/src/server/main.ts:69]: Done.

**Comment by aaronTufts** [comment]
@jasonxh One open question I have on this one that does not apply to Aptos:

Is the `src/client/hooks/useShinamiClient.ts` and `src/client/components/ShinamiClientProvider.tsx` okay to leave in as-is? I just made them as a bare-bones working example of integrating our client on the FE in a method similar to what Mysten does. But if we actually were promoting this or supporting it in a first-class way it might look different. I'm happy to keep it, keep it with a disclaimer, or to remove it for now and use the Mysten client for FE querying for transaction data.

File: aptos/typescript/wallet_adapter_react/src/client/App.tsx (hunk 1 above)
Comment (aaronTufts) [reply aptos/typescript/wallet_adapter_react/src/client/App.tsx]: Fixed

File: aptos/typescript/wallet_adapter_react/src/client/App.tsx (hunk 2 above)
Comment (aaronTufts) [reply aptos/typescript/wallet_adapter_react/src/client/App.tsx]: Done

File: aptos/typescript/wallet_adapter_react/src/server/main.ts (hunk 3 above)
Comment (aaronTufts) [reply aptos/typescript/wallet_adapter_react/src/server/main.ts:75]: done

File: aptos/typescript/wallet_adapter_react/src/client/App.tsx (hunk 1 above)
Comment (jasonxh) [reply aptos/typescript/wallet_adapter_react/src/client/App.tsx]: Still seems off on the catch block. Can you set up your text editor to do it automatically?

File: aptos/typescript/wallet_adapter_react/src/client/main.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import { PontemWallet } from "@pontem/wallet-adapter-plugin";
+import { Network } from "@aptos-labs/ts-sdk";
+
+const wallets = [
+  new PontemWallet()
+];
+
+ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
+  <React.StrictMode>
+            <AptosWalletAdapterProvider
```
Comment (jasonxh) [inline aptos/typescript/wallet_adapter_react/src/client/main.tsx]: Indentation's off here too.

File: aptos/typescript/wallet_adapter_react/src/client/App.tsx (hunk 1 above)
Comment (aaronTufts) [reply aptos/typescript/wallet_adapter_react/src/client/App.tsx]: I just set up VSCode to format on save and I _think_ it's working now.

**Review by jasonxh — APPROVED** [review]

**Pull request merged**